diff --git a/node_modules/hexo-theme-butterfly/layout/archive.pug b/node_modules/hexo-theme-butterfly/layout/archive.pug
index 815cdd1..fa924d9 100644
--- a/node_modules/hexo-theme-butterfly/layout/archive.pug
+++ b/node_modules/hexo-theme-butterfly/layout/archive.pug
@@ -3,7 +3,11 @@ extends includes/layout.pug
 block content
   include ./includes/mixins/article-sort.pug
   #archive
+    #posts-calendar.js-pjax
     - const archiveLength = findArchiveLength(fragment_cache)
     .article-sort-title= _p('page.articles') + ' - ' + archiveLength
     +articleSort(page.posts)
     include includes/pagination.pug
+    //- ad
+    if theme.ad && theme.ad.page
+      .ads-wrap-page!=theme.ad.page
diff --git a/node_modules/hexo-theme-butterfly/layout/category.pug b/node_modules/hexo-theme-butterfly/layout/category.pug
index 234a0af..c1bfb81 100644
--- a/node_modules/hexo-theme-butterfly/layout/category.pug
+++ b/node_modules/hexo-theme-butterfly/layout/category.pug
@@ -12,3 +12,6 @@ block content
       .article-sort-title= _p('page.category') + ' - ' + page.category
       +articleSort(page.posts)
       include includes/pagination.pug
+      //- ad
+      if theme.ad && theme.ad.page
+        .ads-wrap-page!=theme.ad.page
diff --git a/node_modules/hexo-theme-butterfly/layout/includes/head.pug b/node_modules/hexo-theme-butterfly/layout/includes/head.pug
index 0089ad2..cabbc77 100644
--- a/node_modules/hexo-theme-butterfly/layout/includes/head.pug
+++ b/node_modules/hexo-theme-butterfly/layout/includes/head.pug
@@ -15,7 +15,7 @@
 
 meta(charset='UTF-8')
 meta(http-equiv="X-UA-Compatible" content="IE=edge")
-meta(name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0")
+meta(name="viewport" content="width=device-width, initial-scale=1.0, viewport-fit=cover")
 title= tabTitle
 meta(name="author" content=pageAuthor)
 meta(name="copyright" content=pageCopyright)
diff --git a/node_modules/hexo-theme-butterfly/layout/includes/layout.pug b/node_modules/hexo-theme-butterfly/layout/includes/layout.pug
index 6055d1e..bcb7391 100644
--- a/node_modules/hexo-theme-butterfly/layout/includes/layout.pug
+++ b/node_modules/hexo-theme-butterfly/layout/includes/layout.pug
@@ -37,8 +37,7 @@ html(lang=config.language data-theme=theme.display_mode class=htmlClassHideAside
         else
           - var footer_bg = ''
 
-        footer#footer(style=footer_bg)
-          !=partial('includes/footer', {}, {cache: true})
+        footer#footer(style='background: transparent')
 
     else
       include ./404.pug
diff --git a/node_modules/hexo-theme-butterfly/layout/includes/page/categories.pug b/node_modules/hexo-theme-butterfly/layout/includes/page/categories.pug
index 79153c8..7cf828e 100644
--- a/node_modules/hexo-theme-butterfly/layout/includes/page/categories.pug
+++ b/node_modules/hexo-theme-butterfly/layout/includes/page/categories.pug
@@ -1 +1,2 @@
+#categories-chart.js-pjax
 .category-lists!= list_categories()
\ No newline at end of file
diff --git a/node_modules/hexo-theme-butterfly/layout/includes/page/tags.pug b/node_modules/hexo-theme-butterfly/layout/includes/page/tags.pug
index ed7d22c..cdab6ef 100644
--- a/node_modules/hexo-theme-butterfly/layout/includes/page/tags.pug
+++ b/node_modules/hexo-theme-butterfly/layout/includes/page/tags.pug
@@ -1,2 +1,3 @@
+#tags-chart(data-length = "10").js-pjax
 .tag-cloud-list.is-center
   !=cloudTags({source: site.tags, orderby: page.orderby || 'random', order: page.order || 1, minfontsize: 1.2, maxfontsize: 2.1, limit: 0, unit: 'em'})
\ No newline at end of file
diff --git a/node_modules/hexo-theme-butterfly/layout/includes/widget/card_tags.pug b/node_modules/hexo-theme-butterfly/layout/includes/widget/card_tags.pug
index ed09f39..49296b7 100644
--- a/node_modules/hexo-theme-butterfly/layout/includes/widget/card_tags.pug
+++ b/node_modules/hexo-theme-butterfly/layout/includes/widget/card_tags.pug
@@ -9,6 +9,6 @@ if theme.aside.card_tags.enable
       - limit = limit === 0 ? 0 : limit || 40
 
       if theme.aside.card_tags.color
-        .card-tag-cloud!= cloudTags({source: site.tags, orderby: orderby, order: order, minfontsize: 1.15, maxfontsize: 1.45, limit: tagLimit, unit: 'em'})
+        .card-tag-cloud!= cloudTags({source: site.tags, orderby: orderby, order: order, minfontsize: 1.15, maxfontsize: 1.45, limit: limit, unit: 'em'})
       else
         .card-tag-cloud!= tagcloud({orderby: orderby, order: order, min_font: 1.1, max_font: 1.5, amount: limit , color: true, start_color: '#999', end_color: '#99a9bf', unit: 'em'})
diff --git a/node_modules/hexo-theme-butterfly/layout/page.pug b/node_modules/hexo-theme-butterfly/layout/page.pug
index 6c810ef..afe7932 100644
--- a/node_modules/hexo-theme-butterfly/layout/page.pug
+++ b/node_modules/hexo-theme-butterfly/layout/page.pug
@@ -14,7 +14,9 @@ block content
         include includes/page/categories.pug
       default
         include includes/page/default-page.pug
-
     if page.comments !== false && theme.comments && theme.comments.use
       - var commentsJsLoad = true
       !=partial('includes/third-party/comments/index', {}, {cache: true})
+    //- ad
+    if theme.ad && theme.ad.page
+      .ads-wrap-page!=theme.ad.page
diff --git a/node_modules/hexo-theme-butterfly/layout/post.pug b/node_modules/hexo-theme-butterfly/layout/post.pug
index f54ab7a..9ff1cb6 100644
--- a/node_modules/hexo-theme-butterfly/layout/post.pug
+++ b/node_modules/hexo-theme-butterfly/layout/post.pug
@@ -4,7 +4,6 @@ block content
   #post
     if top_img === false
       include includes/header/post-info.pug
-
     article#article-container.post-content!=page.content
     include includes/post/post-copyright.pug
     .tag_share
@@ -29,4 +28,3 @@ block content
     if page.comments !== false && theme.comments && theme.comments.use
       - var commentsJsLoad = true
       !=partial('includes/third-party/comments/index', {}, {cache: true})
-      
\ No newline at end of file
diff --git a/node_modules/hexo-theme-butterfly/layout/tag.pug b/node_modules/hexo-theme-butterfly/layout/tag.pug
index 9f99658..bb26915 100644
--- a/node_modules/hexo-theme-butterfly/layout/tag.pug
+++ b/node_modules/hexo-theme-butterfly/layout/tag.pug
@@ -11,4 +11,6 @@ block content
     #tag
       .article-sort-title= _p('page.tag') + ' - ' + page.tag
       +articleSort(page.posts)
-      include includes/pagination.pug
\ No newline at end of file
+      //- ad
+      if theme.ad && theme.ad.page
+        .ads-wrap-page!=theme.ad.page
diff --git a/node_modules/hexo-theme-butterfly/source/css/_global/index.styl b/node_modules/hexo-theme-butterfly/source/css/_global/index.styl
index a73ae2c..9a6509b 100644
--- a/node_modules/hexo-theme-butterfly/source/css/_global/index.styl
+++ b/node_modules/hexo-theme-butterfly/source/css/_global/index.styl
@@ -44,7 +44,7 @@ body
   font-family: $font-family
   line-height: $text-line-height
   -webkit-tap-highlight-color: rgba(0, 0, 0, 0)
-
+  padding: 0!important
   if !hexo-config('copy.enable')
     user-select: none
 
@@ -79,6 +79,12 @@ if $web-bg
     background-size: cover
     background-repeat: no-repeat
 
+  #web_bg:after
+    content: '';
+    position: absolute;
+    width: 100%;
+    height: 100%;
+    background-color: rgba(0,0,0,.3);
 h1,
 h2,
 h3,
diff --git a/node_modules/hexo-theme-butterfly/source/css/_layout/footer.styl b/node_modules/hexo-theme-butterfly/source/css/_layout/footer.styl
index 55cbc8c..9f69143 100644
--- a/node_modules/hexo-theme-butterfly/source/css/_layout/footer.styl
+++ b/node_modules/hexo-theme-butterfly/source/css/_layout/footer.styl
@@ -5,14 +5,6 @@
   background-position: bottom
   background-size: cover
 
-  if hexo-config('footer_bg') != false
-    &:before
-      position: absolute
-      width: 100%
-      height: 100%
-      background-color: alpha($dark-black, .5)
-      content: ''
-
 #footer-wrap
   position: relative
   padding: 40px 20px
diff --git a/node_modules/hexo-theme-butterfly/source/css/_layout/head.styl b/node_modules/hexo-theme-butterfly/source/css/_layout/head.styl
index ade0db3..7db6383 100644
--- a/node_modules/hexo-theme-butterfly/source/css/_layout/head.styl
+++ b/node_modules/hexo-theme-butterfly/source/css/_layout/head.styl
@@ -19,6 +19,9 @@
     height: $index_top_img_height
     background-attachment: fixed
 
+    &:not(.not-top-img):before {
+      background-color: alpha($dark-black, 0)
+    }
     #site-info
       position: absolute
       top: $index_site_info_top
@@ -289,6 +292,7 @@
       margin-right: 6px
       height: 36px
       vertical-align: middle
+      border-radius: 50%
 
   #toggle-menu
     display: none
diff --git a/node_modules/hexo-theme-butterfly/source/css/_mode/darkmode.styl b/node_modules/hexo-theme-butterfly/source/css/_mode/darkmode.styl
index 680ad3f..a355a06 100644
--- a/node_modules/hexo-theme-butterfly/source/css/_mode/darkmode.styl
+++ b/node_modules/hexo-theme-butterfly/source/css/_mode/darkmode.styl
@@ -14,7 +14,7 @@ if hexo-config('darkmode.enable') || hexo-config('display_mode') == 'dark'
     --tab-botton-color: alpha(#FFFFFF, .7)
     --tab-button-hover-bg: lighten(#121212, 15)
     --tab-button-active-bg: #121212
-    --card-bg: #121212
+    --card-bg: rgba(18,18,18,0.9)
     --sidebar-bg: #121212
     --btn-hover-color: lighten(#121212, 40)
     --btn-color: alpha(#FFFFFF, .7)
@@ -39,15 +39,6 @@ if hexo-config('darkmode.enable') || hexo-config('display_mode') == 'dark'
     --scrollbar-color: lighten(#121212, 5)
     --timeline-bg: lighten(#121212, 5)
 
-    #web_bg:before,
-    #footer:before,
-    #page-header:before
-      position: absolute
-      width: 100%
-      height: 100%
-      background-color: alpha($dark-black, .7)
-      content: ''
-
     #article-container
       code
         background: #2c2c2c
diff --git a/node_modules/hexo-theme-butterfly/source/css/_page/404.styl b/node_modules/hexo-theme-butterfly/source/css/_page/404.styl
index ea8e056..8dfa902 100644
--- a/node_modules/hexo-theme-butterfly/source/css/_page/404.styl
+++ b/node_modules/hexo-theme-butterfly/source/css/_page/404.styl
@@ -12,6 +12,7 @@ if hexo-config('error_404.enable')
 
       .error-content
         @extend .cardHover
+        background: rgba(0, 0, 0, 0)
         overflow: hidden
         margin: 0 20px
         height: 360px
@@ -32,13 +33,14 @@ if hexo-config('error_404.enable')
 
           img
             @extend .imgHover
-            background-color: $theme-color
+            background: var(--card-bg)
 
         .error-info
           display: inline-flex
           flex-direction: column
           justify-content: center
           align-content: center
+          background: var(--card-bg)
           width: 50%
           height: 100%
           vertical-align: top
diff --git a/node_modules/hexo-theme-butterfly/source/css/_page/common.styl b/node_modules/hexo-theme-butterfly/source/css/_page/common.styl
index 3356d17..99b24bd 100644
--- a/node_modules/hexo-theme-butterfly/source/css/_page/common.styl
+++ b/node_modules/hexo-theme-butterfly/source/css/_page/common.styl
@@ -18,7 +18,7 @@
     padding: 20px 5px
 
   +minWidth2000()
-    max-width: 1500px
+    max-width: 1700px
 
   & > div:first-child:not(.recent-posts)
     @extend .cardHover
@@ -58,3 +58,5 @@
   .avatar-img,
   .flink-item-icon
     transform: translateZ(0)
+.ads-wrap-page
+  margin-top: 40px
diff --git a/node_modules/hexo-theme-butterfly/source/css/var.styl b/node_modules/hexo-theme-butterfly/source/css/var.styl
index 04d4326..22e386c 100644
--- a/node_modules/hexo-theme-butterfly/source/css/var.styl
+++ b/node_modules/hexo-theme-butterfly/source/css/var.styl
@@ -32,7 +32,7 @@ $code-font-size = hexo-config('font.code-font-size') ? convert(hexo-config('font
 $font-color = #1F2D3D
 $text-line-height = 2
 $web-bg = hexo-config('background') && unquote(hexo-config('background'))
-$index_top_img_height = hexo-config('index_top_img_height') ? convert(hexo-config('index_top_img_height')) : 100vh
+$index_top_img_height = hexo-config('index_top_img_height') ? convert(hexo-config('index_top_img_height')) : 400px
 $index_site_info_top = hexo-config('index_site_info_top') ? convert(hexo-config('index_site_info_top')) : 43%
 // Global color & SVG
 $light-blue = $theme-color
@@ -43,7 +43,7 @@ $dark-grey = #cacaca
 $white = #FFFFFF
 $whitesmoke = #f5f5f5
 $font-black = #4C4948
-$card-bg = $white
+$card-bg = rgba(255,255,255,0.9)
 $text-highlight-color = $font-color
 $text-hover = $theme-color
 $text-bg-hover = $theme-color
