diff --git a/node_modules/hexo-theme-butterfly/languages/zh-CN.yml b/node_modules/hexo-theme-butterfly/languages/zh-CN.yml
index 9c2919a..d8a9762 100644
--- a/node_modules/hexo-theme-butterfly/languages/zh-CN.yml
+++ b/node_modules/hexo-theme-butterfly/languages/zh-CN.yml
@@ -11,7 +11,7 @@ page:
   articles: 文章总览
   tag: 标签
   category: 分类
-  archives: 归档
+  archives: 时间轴
 
 card_post_count: 条评论
 
@@ -60,7 +60,7 @@ aside:
   card_announcement: 公告
   card_categories: 分类
   card_tags: 标签
-  card_archives: 归档
+  card_archives: 时间轴
   card_recent_post: 最新文章
   card_webinfo:
     headline: 网站资讯
diff --git a/node_modules/hexo-theme-butterfly/languages/zh-TW.yml b/node_modules/hexo-theme-butterfly/languages/zh-TW.yml
index 86623c0..7db9f67 100644
--- a/node_modules/hexo-theme-butterfly/languages/zh-TW.yml
+++ b/node_modules/hexo-theme-butterfly/languages/zh-TW.yml
@@ -11,7 +11,7 @@ page:
   articles: 文章總覽
   tag: 標籤
   category: 分類
-  archives: 歸檔
+  archives: 時間軸
 
 card_post_count: 條評論
 
@@ -60,7 +60,7 @@ aside:
   card_announcement: 公告
   card_categories: 分類
   card_tags: 標籤
-  card_archives: 歸檔
+  card_archives: 時間軸
   card_recent_post: 最新文章
   card_webinfo:
     headline: 網站資訊
diff --git a/node_modules/hexo-theme-butterfly/layout/archive.pug b/node_modules/hexo-theme-butterfly/layout/archive.pug
index 815cdd1..aef2b21 100644
--- a/node_modules/hexo-theme-butterfly/layout/archive.pug
+++ b/node_modules/hexo-theme-butterfly/layout/archive.pug
@@ -3,7 +3,11 @@ extends includes/layout.pug
 block content
   include ./includes/mixins/article-sort.pug
   #archive
+    #posts-calendar.js-pjax
     - const archiveLength = findArchiveLength(fragment_cache)
     .article-sort-title= _p('page.articles') + ' - ' + archiveLength
     +articleSort(page.posts)
     include includes/pagination.pug
+    //- ad
+    if theme.ad && theme.ad.page
+      .ads-wrap-page!=theme.ad.page
\ No newline at end of file
diff --git a/node_modules/hexo-theme-butterfly/layout/includes/page/categories.pug b/node_modules/hexo-theme-butterfly/layout/includes/page/categories.pug
index 79153c8..7cf828e 100644
--- a/node_modules/hexo-theme-butterfly/layout/includes/page/categories.pug
+++ b/node_modules/hexo-theme-butterfly/layout/includes/page/categories.pug
@@ -1 +1,2 @@
+#categories-chart.js-pjax
 .category-lists!= list_categories()
\ No newline at end of file
diff --git a/node_modules/hexo-theme-butterfly/layout/includes/page/tags.pug b/node_modules/hexo-theme-butterfly/layout/includes/page/tags.pug
index ed7d22c..cdab6ef 100644
--- a/node_modules/hexo-theme-butterfly/layout/includes/page/tags.pug
+++ b/node_modules/hexo-theme-butterfly/layout/includes/page/tags.pug
@@ -1,2 +1,3 @@
+#tags-chart(data-length = "10").js-pjax
 .tag-cloud-list.is-center
   !=cloudTags({source: site.tags, orderby: page.orderby || 'random', order: page.order || 1, minfontsize: 1.2, maxfontsize: 2.1, limit: 0, unit: 'em'})
\ No newline at end of file
diff --git a/node_modules/hexo-theme-butterfly/layout/includes/widget/card_tags.pug b/node_modules/hexo-theme-butterfly/layout/includes/widget/card_tags.pug
index ed09f39..49296b7 100644
--- a/node_modules/hexo-theme-butterfly/layout/includes/widget/card_tags.pug
+++ b/node_modules/hexo-theme-butterfly/layout/includes/widget/card_tags.pug
@@ -9,6 +9,6 @@ if theme.aside.card_tags.enable
       - limit = limit === 0 ? 0 : limit || 40
 
       if theme.aside.card_tags.color
-        .card-tag-cloud!= cloudTags({source: site.tags, orderby: orderby, order: order, minfontsize: 1.15, maxfontsize: 1.45, limit: tagLimit, unit: 'em'})
+        .card-tag-cloud!= cloudTags({source: site.tags, orderby: orderby, order: order, minfontsize: 1.15, maxfontsize: 1.45, limit: limit, unit: 'em'})
       else
         .card-tag-cloud!= tagcloud({orderby: orderby, order: order, min_font: 1.1, max_font: 1.5, amount: limit , color: true, start_color: '#999', end_color: '#99a9bf', unit: 'em'})
diff --git a/node_modules/hexo-theme-butterfly/layout/page.pug b/node_modules/hexo-theme-butterfly/layout/page.pug
index 6c810ef..59a3418 100644
--- a/node_modules/hexo-theme-butterfly/layout/page.pug
+++ b/node_modules/hexo-theme-butterfly/layout/page.pug
@@ -14,7 +14,10 @@ block content
         include includes/page/categories.pug
       default
         include includes/page/default-page.pug
-
+    <script async="async" data-cfasync="false" src="//pl18904263.highrevenuegate.com/60a9585e87207b9db130c85b2cfbff86/invoke.js"></script><div id="container-60a9585e87207b9db130c85b2cfbff86"></div>
     if page.comments !== false && theme.comments && theme.comments.use
       - var commentsJsLoad = true
       !=partial('includes/third-party/comments/index', {}, {cache: true})
+    //- ad
+    if theme.ad && theme.ad.page
+      .ads-wrap-page!=theme.ad.page
\ No newline at end of file
diff --git a/node_modules/hexo-theme-butterfly/layout/post.pug b/node_modules/hexo-theme-butterfly/layout/post.pug
index f54ab7a..9ff1cb6 100644
--- a/node_modules/hexo-theme-butterfly/layout/post.pug
+++ b/node_modules/hexo-theme-butterfly/layout/post.pug
@@ -4,7 +4,6 @@ block content
   #post
     if top_img === false
       include includes/header/post-info.pug
-
     article#article-container.post-content!=page.content
     include includes/post/post-copyright.pug
     .tag_share
@@ -29,4 +28,3 @@ block content
     if page.comments !== false && theme.comments && theme.comments.use
       - var commentsJsLoad = true
       !=partial('includes/third-party/comments/index', {}, {cache: true})
-      
\ No newline at end of file
diff --git a/node_modules/hexo-theme-butterfly/source/css/_layout/head.styl b/node_modules/hexo-theme-butterfly/source/css/_layout/head.styl
index ade0db3..8ca7c24 100644
--- a/node_modules/hexo-theme-butterfly/source/css/_layout/head.styl
+++ b/node_modules/hexo-theme-butterfly/source/css/_layout/head.styl
@@ -289,6 +289,7 @@
       margin-right: 6px
       height: 36px
       vertical-align: middle
+      border-radius: 50%
 
   #toggle-menu
     display: none
diff --git a/node_modules/hexo-theme-butterfly/source/css/_page/common.styl b/node_modules/hexo-theme-butterfly/source/css/_page/common.styl
index 3356d17..99b24bd 100644
--- a/node_modules/hexo-theme-butterfly/source/css/_page/common.styl
+++ b/node_modules/hexo-theme-butterfly/source/css/_page/common.styl
@@ -18,7 +18,7 @@
     padding: 20px 5px
 
   +minWidth2000()
-    max-width: 1500px
+    max-width: 1700px
 
   & > div:first-child:not(.recent-posts)
     @extend .cardHover
@@ -58,3 +58,5 @@
   .avatar-img,
   .flink-item-icon
     transform: translateZ(0)
+.ads-wrap-page
+  margin-top: 40px
