diff --git a/node_modules/hexo-theme-butterfly/languages/zh-CN.yml b/node_modules/hexo-theme-butterfly/languages/zh-CN.yml
index 9c2919a..d8a9762 100644
--- a/node_modules/hexo-theme-butterfly/languages/zh-CN.yml
+++ b/node_modules/hexo-theme-butterfly/languages/zh-CN.yml
@@ -11,7 +11,7 @@ page:
   articles: 文章总览
   tag: 标签
   category: 分类
-  archives: 归档
+  archives: 时间轴
 
 card_post_count: 条评论
 
@@ -60,7 +60,7 @@ aside:
   card_announcement: 公告
   card_categories: 分类
   card_tags: 标签
-  card_archives: 归档
+  card_archives: 时间轴
   card_recent_post: 最新文章
   card_webinfo:
     headline: 网站资讯
diff --git a/node_modules/hexo-theme-butterfly/languages/zh-TW.yml b/node_modules/hexo-theme-butterfly/languages/zh-TW.yml
index 86623c0..7db9f67 100644
--- a/node_modules/hexo-theme-butterfly/languages/zh-TW.yml
+++ b/node_modules/hexo-theme-butterfly/languages/zh-TW.yml
@@ -11,7 +11,7 @@ page:
   articles: 文章總覽
   tag: 標籤
   category: 分類
-  archives: 歸檔
+  archives: 時間軸
 
 card_post_count: 條評論
 
@@ -60,7 +60,7 @@ aside:
   card_announcement: 公告
   card_categories: 分類
   card_tags: 標籤
-  card_archives: 歸檔
+  card_archives: 時間軸
   card_recent_post: 最新文章
   card_webinfo:
     headline: 網站資訊
diff --git a/node_modules/hexo-theme-butterfly/layout/archive.pug b/node_modules/hexo-theme-butterfly/layout/archive.pug
index 815cdd1..f04dbc5 100644
--- a/node_modules/hexo-theme-butterfly/layout/archive.pug
+++ b/node_modules/hexo-theme-butterfly/layout/archive.pug
@@ -3,6 +3,7 @@ extends includes/layout.pug
 block content
   include ./includes/mixins/article-sort.pug
   #archive
+    #posts-calendar.js-pjax
     - const archiveLength = findArchiveLength(fragment_cache)
     .article-sort-title= _p('page.articles') + ' - ' + archiveLength
     +articleSort(page.posts)
diff --git a/node_modules/hexo-theme-butterfly/layout/includes/page/categories.pug b/node_modules/hexo-theme-butterfly/layout/includes/page/categories.pug
index 79153c8..d440a28 100644
--- a/node_modules/hexo-theme-butterfly/layout/includes/page/categories.pug
+++ b/node_modules/hexo-theme-butterfly/layout/includes/page/categories.pug
@@ -1 +1,2 @@
+#categories-chart.js-pjax
 .category-lists!= list_categories()
diff --git a/node_modules/hexo-theme-butterfly/layout/includes/page/tags.pug b/node_modules/hexo-theme-butterfly/layout/includes/page/tags.pug
index ed7d22c..d2a2556 100644
--- a/node_modules/hexo-theme-butterfly/layout/includes/page/tags.pug
+++ b/node_modules/hexo-theme-butterfly/layout/includes/page/tags.pug
@@ -1,2 +1,3 @@
+#tags-chart(data-length = "10").js-pjax
 .tag-cloud-list.is-center
   !=cloudTags({source: site.tags, orderby: page.orderby || 'random', order: page.order || 1, minfontsize: 1.2, maxfontsize: 2.1, limit: 0, unit: 'em'})
diff --git a/node_modules/hexo-theme-butterfly/layout/includes/widget/card_announcement.pug b/node_modules/hexo-theme-butterfly/layout/includes/widget/card_announcement.pug
index 9e63627..c1115a2 100644
--- a/node_modules/hexo-theme-butterfly/layout/includes/widget/card_announcement.pug
+++ b/node_modules/hexo-theme-butterfly/layout/includes/widget/card_announcement.pug
@@ -4,3 +4,4 @@ if theme.aside.card_announcement.enable
       i.fas.fa-bullhorn.fa-shake
       span= _p('aside.card_announcement')
     .announcement_content!= theme.aside.card_announcement.content
+      .twopeople!= '<div class="twopeople"><div class="container" style="height:200px;"><canvas class="illo" width="800" height="800" style="max-width: 200px; max-height: 200px; touch-action: none; width: 640px; height: 640px;"></canvas></div><script src="https://cdn.jsdelivr.net/gh/Justlovesmile/CDN/js/twopeople1.js"></script><script src="https://cdn.jsdelivr.net/gh/Justlovesmile/CDN/js/zdog.dist.js"></script><script id="rendered-js" src="https://cdn.jsdelivr.net/gh/Justlovesmile/CDN/js/twopeople.js"></script><style>.twopeople{margin: 0;align-items: center;justify-content: center;text-align: center;}canvas {display: block;margin: 0 auto;cursor: move;}</style></div>'
\ No newline at end of file
diff --git a/node_modules/hexo-theme-butterfly/layout/post.pug b/node_modules/hexo-theme-butterfly/layout/post.pug
index f54ab7a..b8c165f 100644
--- a/node_modules/hexo-theme-butterfly/layout/post.pug
+++ b/node_modules/hexo-theme-butterfly/layout/post.pug
@@ -4,8 +4,11 @@ block content
   #post
     if top_img === false
       include includes/header/post-info.pug
-
-    article#article-container.post-content!=page.content
+    article#article-container
+      //- ad
+      if theme.ad && theme.ad.post
+        .ads-wrap-top!=theme.ad.post
+      .post-content!=page.content
     include includes/post/post-copyright.pug
     .tag_share
       if (theme.post_meta.post.tags)
@@ -19,7 +22,7 @@ block content
 
     //- ad
     if theme.ad && theme.ad.post
-      .ads-wrap!=theme.ad.post
+      .ads-wrap-bottom!=theme.ad.post
 
     if theme.post_pagination
       include includes/pagination.pug
@@ -29,4 +32,3 @@ block content
     if page.comments !== false && theme.comments && theme.comments.use
       - var commentsJsLoad = true
       !=partial('includes/third-party/comments/index', {}, {cache: true})
-      
\ No newline at end of file
diff --git a/node_modules/hexo-theme-butterfly/source/css/_layout/head.styl b/node_modules/hexo-theme-butterfly/source/css/_layout/head.styl
index ade0db3..8ca7c24 100644
--- a/node_modules/hexo-theme-butterfly/source/css/_layout/head.styl
+++ b/node_modules/hexo-theme-butterfly/source/css/_layout/head.styl
@@ -289,6 +289,7 @@
       margin-right: 6px
       height: 36px
       vertical-align: middle
+      border-radius: 50%
 
   #toggle-menu
     display: none
diff --git a/node_modules/hexo-theme-butterfly/source/css/_layout/post.styl b/node_modules/hexo-theme-butterfly/source/css/_layout/post.styl
index 4915ee6..085c632 100644
--- a/node_modules/hexo-theme-butterfly/source/css/_layout/post.styl
+++ b/node_modules/hexo-theme-butterfly/source/css/_layout/post.styl
@@ -250,6 +250,7 @@ beautify()
         color: $noticeOutdate-border
         content: '\f071'
         transform: translateY(-50%)
-
-  .ads-wrap
+  .ads-wrap-top
+    margin-bottom: 40px 0
+  .ads-wrap-bottom
     margin: 40px 0
